eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -91,209 +96,183 @@ object Graph {
+      val initialSize = 100
+      val bestWeights = new java.util.HashMap[PublicKey, RichWeight](initialSize)
+      val bestEdges = new java.util.HashMap[PublicKey, GraphEdge](initialSize)
+      val toExplore = new org.jheaps.tree.SimpleFibonacciHeap[WeightedNode, Short](NodeComparator)
+
+      // initialize the queue and cost array with the initial weight
+      bestWeights.put(targetNode, initialWeight)
+      toExplore.insert(WeightedNode(targetNode, initialWeight))
+
+      var targetFound = false
+      while (!toExplore.isEmpty && !targetFound) {
+        // node with the smallest distance from the target
+        val current = toExplore.deleteMin().getKey // O(log(n))
+        if (current.key != sourceNode) {
+          val currentWeight = bestWeights.get(current.key) // NB: there is always an entry for the current in the 'weights' map
+          // build the neighbors with optional extra edges
+          val currentNeighbors = extraEdges.isEmpty match {
+            case true => g.getIncomingEdgesOf(current.key)
+            case false =>
+              val extraNeighbors = extraEdges.filter(_.desc.b == current.key)
+              // the resulting set must have only one element per shortChannelId; we prioritize extra edges
+              g.getIncomingEdgesOf(current.key).filterNot(e => extraNeighbors.exists(_.desc.shortChannelId == e.desc.shortChannelId)) ++ extraNeighbors
+          }
+          currentNeighbors.foreach { edge =>
+            val neighbor = edge.desc.a
+            // NB: this contains the amount (including fees) that will need to be sent to `neighbor`, but the amount that
+            // will be relayed through that edge is the one in `currentWeight`.
+            val neighborWeight = addEdgeWeight(sender, edge, currentWeight, currentBlockHeight, wr)
+            val canRelayAmount = currentWeight.cost <= edge.capacity &&
+              edge.balance_opt.forall(currentWeight.cost <= _) &&
+              edge.update.htlcMaximumMsat.forall(currentWeight.cost <= _) &&
+              currentWeight.cost >= edge.update.htlcMinimumMsat
+            if (canRelayAmount && boundaries(neighborWeight) && !ignoredEdges.contains(edge.desc) && !ignoredVertices.contains(neighbor)) {
+              // we don't use "getOrDefault" because it is not available in JDK7
+              val previousNeighborWeight = bestWeights.get(neighbor) match {
+                case null => RichWeight(MilliSatoshi(Long.MaxValue), Int.MaxValue, CltvExpiryDelta(Int.MaxValue), Double.MaxValue)
+                case w => w
+              }
+              // if this path between neighbor and the target has a shorter distance than previously known, we select it
+              if (neighborWeight.weight < previousNeighborWeight.weight) {
+                // update the best edge for this vertex
+                bestEdges.put(neighbor, edge)
+                // continue exploring from this updated node
```

Comment:
Disagree :) , adding the new weight for this node won't necessarily result in continuing the exploration of the search space from it, the next iteration will pop off the queue (which is actually a fibonacci heap) the node with the smallest distance from the target and it might not be `neighbor`

Reply:
Good catch, that comment isn't accurate

Reply:
Fixed in https://github.com/ACINQ/eclair/pull/1395/commits/702493f329b4f60800fb20054bd835df40b16264